### PR TITLE
Om 1572 scan request

### PR DESCRIPTION
**Pull request by emessiha**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -2,43 +2,57 @@
...
-    var ex = new extractor.Extractor(configuration.projectDir);
-    ex.writeProjectMeta(meta);
-    ex.doTrap();
-    await ex.renameTrap();
-    const compressedFile = await ex.compressOutput();
-
-    const sendMechanism = new transport.Transport();
-    sendMechanism.sendResult(
+    var branch = meta.remotes
+    if (!!!branch) {
```
Comment (ohamel-softwaresecure) [inline main.js:23]: wha?

File: main.js [hunk 2]
```diff
@@ -2,43 +2,57 @@
...
+       console.error('ERROR: Git project does not seem to have remote branch tracking. Unable to proceed with scan.');
+       exit(1);
+    }
+    var gitStatus = await gitInstance.getGitStatus();
+    if (!gitStatus.isClean()) {
+        console.error('ERROR: Git project seems to have local changes or is not clean. Make sure local copy is in sync with remote.');
+        exit(1);
+    }
+    if (gitStatus.ahead > 0 || gitStatus.behind > 0) {
+        console.error('ERROR: Git project seems to be out of sync with remote (ahead/behind). Make sure local copy is in sync with remote.');
```
Comment (ohamel-softwaresecure) [inline main.js:34]: does this fetch to obtain the latest info?

File: main.js [hunk 3]
```diff
@@ -2,43 +2,57 @@
...
+        exit(1);
+    }
+    var projectProviderUrl = repoInfo.href;
+    
+    var start = new Date();
+    console.log(util.format('RESHIFT SECURITY: Starting security scan for project %s', repoInfo.full_name));
```
Comment (ohamel-softwaresecure) [inline main.js:40]: s/security scan/scan

File: main.js (hunk 1 above)
Comment (emessiha) [reply main.js:23]: Lol this is short hand for not null, not undefined and not empty :) apparently in javascript they are different lol

**Pull request closed without merging**